### PR TITLE
Update supported WP version to 5.7

### LINKED ISSUE #45: Are we compatible with WordPress 5.7?
Reported by @lolzim 

[WordPress 5.7](https://make.wordpress.org/core/2021/02/23/wordpress-5-7-field-guide/) is coming soon, so we should check that nothing in the Plugin will break as a result.

### PR DESCRIPTION
**Pull request by Rumperuu**
Close #45

### CODE REVIEW

**Review by pewgeuges — COMMENTED** [review]
We need to add the Package Version field in main.php too.

File: footnotes.php [hunk 1]
```diff
@@ -16,38 +18,39 @@
...
- * Tagged Version    Refers to the latest tag.
+ * Version           Parsed in the main PHP file header for update configuration.
+ *                   Refers to the latest tag.
  *                   Typically the best available version.
  *                   May not be released, due to release frequency concerns.
  *
- * Stable Tag        Parsed in trunk/ for release configuration.
- *                   Must not be greater than the Version in MAIN.php header.
+ * Stable Tag        Parsed in the readme.txt in trunk/ for release configuration.
+ *                   May be smaller than Version, avoiding too frequent releases.
```
Comment (pewgeuges) [inline footnotes.php:53]: Previously, this line referred to the idea that the Stable Tag enters into the comparison.

That has been [unveiled as a misconception on March 11](https://github.com/markcheret/footnotes/pull/62#pullrequestreview-610365471).

**Review by pewgeuges — COMMENTED** [review]
Needed to eliminate uninformed comments made while believing [a misleading post from WP](https://meta.trac.wordpress.org/ticket/5645#comment:4).

File: readme.txt [hunk 2]
```diff
@@ -2,10 +2,10 @@
 Contributors: mark.cheret, lolzim, rumperuu, aricura, misfist, ericakfranz, dartiss, docteurfitness, felipelavinz, martinneumannat, matkus2, meglio, spaceling, vonpiernik, pewgeuges
 Tags: footnote, footnotes, bibliography, formatting, notes, Post, posts, reference, referencing
 Requires at least: 3.9
-Tested up to: 5.6.1
+Tested up to: 5.7
 Requires PHP: 5.6
-Package Version: 2.5.11d1
-Tagged Version: 2.5.10
+Package Version: 2.5.11d2
```
Comment (pewgeuges) [inline readme.txt:7]: Incremented d version as SVN trunk/ has become 2.5.11d1 after updating the tested-up-to value.

File: readme.txt [hunk 3]
```diff
@@ -2,10 +2,10 @@
...
+Version: 2.5.10
```
Comment (pewgeuges) [inline readme.txt:8]: Synced the Version field label (previously Tagged Version as opposed to Package Version) with its counterpart in main.php.

File: readme.txt [hunk 4]
```diff
@@ -82,7 +82,7 @@ Visit this swift write-up from a **footnotes** user by the name of **Southwest**
 
 = 2.5.11 =
 - Adding: Documentation: Readme.txt: comment line below the 'Stable Tag' field to warn that this is (unexpectedly) parsed for release configuration.
-- Adding: Documentation: Readme.txt: informative 'Tagged Version' field pointing a tag folder in SVN for bugfix versions available ahead of the stable tag.
+- Adding: Documentation: Readme.txt: informative 'Version' field pointing a tag folder in SVN for bugfix versions available ahead of the stable tag.
```
Comment (pewgeuges) [inline readme.txt:85]: Synced the related changelog item after renaming the label.

**Review by pewgeuges — APPROVED** [review]
Getting rid of misleading new label I used while not realising the equivalence.

**Review by markcheret — APPROVED** [review]
Looking good. Thanks for your work.

**Comment by pewgeuges** [comment]
@Rumperuu 

> > The `compliance` label was added for some reason, as we should be careful to stay compliant when setting up processes while we don’t control how WordPress manages the plugin ecosystem.
> 
> Equally, it was removed for a reason. This PR has nothing to do with adherence to the WP Coding Standards.

Sorry, I forgot about the label’s narrow scope and overstretched its semantics. Thanks for removing it!

**Pull request merged**